packages/file-store/configstores/FileConfigstore.ts

```diff
@@ -37,13 +37,12 @@ export class FileConfigstore implements Configstore {
 
   async list(): Promise<Array<string>> {
     return this.queue.add(async () => {
-      const files = await fs.readdir(this.directory, {withFileTypes: true})
-      const promises = files
-        .filter((file) => file.isFile() && file.name.endsWith('.json'))
-        .map((file) => fs.readFile(path.resolve(file.path, file.name), 'utf8'))
-
-      return Promise.all(promises)
-    }) as Promise<string[]>
+      const files = await fs.readdir(this.directory)
+      // list is expected to return an array of IDs,
+      // which is the same as the file name.
+      // But the JSON info files are also in the directory so we filter those.
+      return files.filter((file) => !file.endsWith('.json'))
```

Comment:
I'd check the opposite and consider only files that have JSON info file present. This way you ignore any random files in this folder.